ui/controls.js

```diff
@@ -522,6 +522,29 @@ shaka.ui.Controls = class extends shaka.util.FakeEventTarget {
     return this.seekBar_ ? this.seekBar_.getValue() : this.video_.currentTime;
   }
 
+  /**
+   * Checking the efficiency of a video
+   * @return {?number}
+   * @export
+   */
+  getBufferFillPercentage() {
+    const bufferedLength_ = this.video_.buffered.length;
+    const bufferedEnd_ =
+        bufferedLength_ ? this.video_.buffered.end(bufferedLength_ - 1) : 0;
+    const bufferingGoal_ = this.video_.currentTime +
+        Math.min(this.player_.getConfiguration().streaming.bufferingGoal,
+            (this.player_.seekRange().end - this.video_.currentTime));
+
+    if (bufferedEnd_ >= bufferingGoal_) {
+      return 100;
+    } else if (bufferedEnd_ < bufferingGoal_) {
+      return ((bufferedEnd_/bufferingGoal_)*100);
```

Comment:
This calculation is wrong, because it doesn't account for the starting point.  The start of the range should be currentTime, not 0.

Reply:
@joeyparrish the formula of bufferingGoal_ does account for the starting point. I named the variable wrong, I'll try coming up with a better name. I am bad at it so I am open to suggestions.

Reply:
I think the formula is wrong . The fraction should be between buffering goal and the length buffered after the current time . So the formula will be ` (lengthToBeBuffered-currentTime)/bufferingGoal `